presto-main/src/main/java/com/facebook/presto/execution/StageExecutionInfo.java

```diff
@@ -309,4 +210,172 @@ public static StageExecutionInfo unscheduledExecutionInfo(int stageId, boolean i
+
+        public void processTaskStats(TaskStats taskStats)
+        {
+            totalDrivers += taskStats.getTotalDrivers();
+            queuedDrivers += taskStats.getQueuedDrivers();
+            runningDrivers += taskStats.getRunningDrivers();
+            blockedDrivers += taskStats.getBlockedDrivers();
+            completedDrivers += taskStats.getCompletedDrivers();
+
+            cumulativeUserMemory += taskStats.getCumulativeUserMemory();
+            cumulativeTotalMemory += taskStats.getCumulativeTotalMemory();
+
+            long taskUserMemory = taskStats.getUserMemoryReservationInBytes();
+            long taskSystemMemory = taskStats.getSystemMemoryReservationInBytes();
+            userMemoryReservation += taskUserMemory;
+            totalMemoryReservation += taskUserMemory + taskSystemMemory;
+
+            totalScheduledTime += taskStats.getTotalScheduledTimeInNanos();
+            totalCpuTime += taskStats.getTotalCpuTimeInNanos();
+            totalBlockedTime += taskStats.getTotalBlockedTimeInNanos();
+
+            totalAllocation += taskStats.getTotalAllocationInBytes();
+
+            rawInputDataSize += taskStats.getRawInputDataSizeInBytes();
+            rawInputPositions += taskStats.getRawInputPositions();
+
+            processedInputDataSize += taskStats.getProcessedInputDataSizeInBytes();
+            processedInputPositions += taskStats.getProcessedInputPositions();
+
+            outputDataSize += taskStats.getOutputDataSizeInBytes();
+            outputPositions += taskStats.getOutputPositions();
+
+            physicalWrittenDataSize += taskStats.getPhysicalWrittenDataSizeInBytes();
+
+            fullGcCount += taskStats.getFullGcCount();
+            fullGcTaskCount += taskStats.getFullGcCount() > 0 ? 1 : 0;
+
+            int gcSec = toIntExact(MILLISECONDS.toSeconds(taskStats.getFullGcTimeInMillis()));
+            totalFullGcSec += gcSec;
+            minFullGcSec = min(minFullGcSec, gcSec);
+            maxFullGcSec = max(maxFullGcSec, gcSec);
+
+            updateOperatorStats(taskStats);
+            updateRuntimeStats(taskStats);
+        }
+
+        private void updateOperatorStats(TaskStats taskStats)
+        {
+            // First collect all operator stats by their ID
+            Map<String, List<OperatorStats>> operatorStatsById = new HashMap<>();
+            for (PipelineStats pipeline : taskStats.getPipelines()) {
+                for (OperatorStats operatorStats : pipeline.getOperatorSummaries()) {
+                    String id = pipeline.getPipelineId() + "." + operatorStats.getOperatorId();
+                    operatorStatsById.computeIfAbsent(id, k -> new ArrayList<>()).add(operatorStats);
+                }
+            }
+            // Merge each group into a single result
+            operatorStatsById.forEach((id, statsToMerge) -> statsToMerge.stream().reduce(OperatorStats::add).ifPresent(merged -> operatorToStats.put(id, merged)));
```

Comment:
Ideally you would want to merge once operator stats are collected from all tasks (e.g.: in `getOperatorToStatsById`)

Basically the idea is to get rid of [this method](https://github.com/prestodb/presto/blob/c2b22c097aa366ed446caa84c7f83a8ffad1bb96/presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java#L719) and always use [this one](https://github.com/prestodb/presto/blob/c2b22c097aa366ed446caa84c7f83a8ffad1bb96/presto-main/src/main/java/com/facebook/presto/operator/OperatorStats.java#L724)